src/main/java/com/google/cloud/pubsublite/spark/LimitingHeadOffsetReader.java

```diff
@@ -96,5 +99,6 @@ public PslSourceOffset getHeadOffset() {
   @Override
   public void close() {
     topicStatsClient.close();
+    partitionCountReader.close();
```

Comment:
Yes, we won't close the 2nd one if 1st one already fails closing. But I don't see too much point in ensuring partitionCountReader to be closed since it doesn't do extra state persistence other than closing an admin client. The only other thing I see might be worthy is to add a try catch on top of both so any exception in close() won't impact other parts of the task, however, the close() happens when the whole spark task finish, so we should be fine throwing exception here.

Reply:
Done to attempt closing all.